test/arrays/views/viewPreserveDist.bad

```diff
@@ -1,4 +1,20 @@
-0.0 0.1 0.1 0.2 0.2 0.3 0.3
-0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0
-0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0
-0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0
+D is: {1..8}
+X is: 0.0 0.0 0.1 0.1 0.2 0.2 0.3 0.3
+Y is: 0.0 0.0 0.1 0.1 0.2 0.2 0.3 0.3
+
+D is: {2..8}
+X is: 0.0 0.1 0.1 0.2 0.2 0.3 0.3
+Y is: 0.0 0.1 0.1 0.2 0.2 0.3 0.3
+
+D is: {1..8}
+X is: 0.0 0.0 0.0 0.0 0.2 0.2 0.2 0.2
+Y is: 0.0 0.0 0.0 0.0 0.2 0.2 0.2 0.2
+
+D is: {1..8}
+X is: 0.2 0.2 0.2 0.2 0.3 0.3 0.3 0.3
+Y is: 0.2 0.2 0.2 0.2 0.3 0.3 0.3 0.3
+
+D is: {0..15 by 2}
+X is: 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0
+Y is: 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0
+
```

Comment:
I think I see what's going on here. The rank-change cases are now working, but the reindex case is not. I think it would make sense to split this test into a separate reindex-preserves-distribution future and let the rank-change cases be a normal test (that now passes, thanks to this PR).